### ISSUE TITLE
git-crypt can't handle utf-8

### ISSUE THREAD

**Issue by Crims-on** [AGWA/git-crypt#298]
\> git-crypt unlock ../key
git-crypt: System error: CreateFileA: - 2 - Omr├Ñden/- 2 - Omr├Ñden.mdThe system cannot find the path specified.

**Comment by alerque** [comment]
I use UTF8 both in files and filenames all the time. What exactly led up to this? What system was the locked file created on, what filesystem are you decoding on, etc.?

**Comment by alerque** [comment]
I assume you mean "Git repository" not "github", but whatever...

Was the first machine you used to encrypt and push the repository a similar Windows machine? Can you clone lock/unlock the repository from the first machine? Can you clone it to a different location and unlock also from the first machine?

**Comment by AGWA** [comment]
Yes, `get_file_attributes` should indeed be passing the `-z` option. Good catch!

@Crims-on - if you could create a repo on GitHub containing the problematic filenames so that I can reproduce the problem, it would help me fix the bug.

**Comment by bxm156** [comment]
I encountered this issue today as well, likely because one of my files had an emoji in the filename. If ya'll still need a example repo, let me know and I can probably share something from mine.

**Comment by AGWA** [comment]
@bxm156 yes, an example repo would still be very useful, thank you!

**Comment by AGWA** [comment]
Community help wanted: a Git repository containing a filename that demonstrates this bug.

**Comment by linearizability** [comment]
> Community help wanted: a Git repository containing a filename that demonstrates this bug.


@AGWA 
Hello, I can provide a sample repository. I submitted it along with the key

git@github.com:linearizability/git-crypt-error-demo.git

The reason is likely due to the directory name being in Chinese

<img width="1443" height="88" alt="Image" src="https://github.com/user-attachments/assets/88f03389-7da7-4991-970d-4ad92c7e48cc" />

**Comment by reneleonhardt** [comment]
Thank you very much @linearizability here is the web link for anyone interested to try out this bug.
https://github.com/linearizability/git-crypt-error-demo
